lib/puppet/util/execution.rb

```diff
@@ -114,6 +114,8 @@ def self.execfail(command, exception)
   #   an Array the first element should be the executable and the rest of the
   #   elements should be the individual arguments to that executable.
   # @param options [Hash] a Hash of options
+  # @option options [String] :cwd the directory from which to run the command. Defaults to the current directory.
+  #   Raises an error if the directory does not exist.
```

Comment:
If this option is only intended to be used on the agent in types and providers then here, and maybe the commit message for 9a34d94f6e34c29d14fd8ff757712d8c208382ad, are probably the best places to mention that.

Reply:
Updated the commit message + those comments. Hopefully that's clear now.